### ISSUE TITLE
Pulsar 2.7.1 checks schema compatibility even if set to ALWAYS_COMPATIBLE

### ISSUE THREAD

**Issue by gvolpe** [apache/pulsar#10244]
**Describe the bug**

Pulsar checks schema compatibility of JSON schemas when deploying new consumers even when the schema compatibility strategy of the namespace is set to [ALWAYS_COMPATIBLE](https://pulsar.apache.org/docs/en/schema-evolution-compatibility/#schema-compatibility-check-strategy), where *all changes should be allowed*.

**To Reproduce**

1. Set the schema compatibility strategy to `FULL` (the default for JSON and AVRO).
2. Deploy a producer and consumer with any JSON schema.
3. Set the schema compatibility strategy to `ALWAYS_COMPATIBLE`.
4. Deploy new versions of the producer and consumer with breaking changes in the schema, or even a new schema.

At this point, Pulsar will fail with an error similar to this:

```
[pulsar-client-io-1-1] WARN  o.a.p.c.i.ConsumerImpl - [non-persistent://public/default/test][test-sub] Failed to subscribe to topic on IP:PORT
org.apache.pulsar.client.api.PulsarClientException$IncompatibleSchemaException: org.apache.avro.SchemaValidationException: Unable to read schema:
```

To set the schema compatibility strategy of a namespace, run the following command.

```
# pulsar-admin namespaces set-schema-compatibility-strategy --compatibility ALWAYS_COMPATIBLE public/default
```

**Expected behavior**

Pulsar should NOT check for schema compatibility when the strategy is set to `ALWAYS_COMPATIBLE`.

**Comment by codelipenghui** [comment]
@gvolpe Looks the problem is not about the compatibility, could you please share the schema definition? From the log, seem the problem is the the provided schema definition is not valid.

**Comment by gvolpe** [comment]
@codelipenghui here's the full error including the old schema (which was a dummy one)

```
[pulsar-client-io-1-1] WARN  o.a.p.c.i.ClientCnx - [id: 0x6027a271, L:/172.17.101.212:43656 - R:172.25.24.247/172.25.24.247:6650] Received error from server: org.apache.avro.SchemaValidationException: Unable to read schema:
{
  "type" : "record",
  "name" : "Event",
  "namespace" : "cr.domain.shared",
  "fields" : [ ]
}
using schema:
{
  "type" : "record",
  "name" : "Event",
  "namespace" : "cr.domain.shared",
  "fields" : [ ]
}
```

The system was working just fine with this until we released a new update of the schema. I think the real problem here is that Pulsar should not verify the schema at all with the selected compatibility strategy.

**Comment by eolivelli** [comment]
@gvolpe do you have a chance to try out current master ? 2.8.0-SNAPSHOT
we fixed as couple of problems recently

**Comment by eolivelli** [comment]
it this fixed by #10386 ?

**Comment by codelipenghui** [comment]
Yes, I think this can be fixed by #10386

**Comment by gvolpe** [comment]
Sorry I didn't get to reply on time. When I checked, I saw the ticket was already closed.

FWIW there still seem to be a bug in Pulsar. We have the schema compatibility strategy set to `ALWAYS_COMPATIBLE` but Pulsar still tries to validate the schema. This is the error we get now with Pulsar 2.7.2:

```
Topic does not have schema to check
```

I wish I had more but this is all I got for now. I'll share more if I get more information.
